### DISCUSSION TITLE
Keeping only recent data and forecasts

### DISCUSSION THREAD

**Discussion by philipredstone** [open-meteo/open-meteo#385] [Q&A] [answered]
First off, gotta say I'm really digging the Open-Meteo project. I've been running it myself and it's been fantastic.

I've noticed, though, it's storing all the data it collects indefinitely. While that's neat and all, I'm finding I don't really need to keep everything. 

What I'm looking for is a way to keep just the data from the last 7 days and the forecasts, and let the rest go. Is there a setting or a trick to make that happen? It'd make managing my storage a lot easier.

Any advice you've got would be super helpful.

Thanks!

**Comment by patrick-zippenfenig** [comment, marked as answer]
I typically use 2 cronjobs for cleanup. Data on pressure level get deleted after 15 days while other surface data remains for 100 days after last modification.

```
0 * * * * find /var/lib/openmeteo-api/data/ -type f -wholename "/var/lib/openmeteo-api/data/omfile-*" -name "*hPa*" -mtime +15 -delete
5 * * * * find /var/lib/openmeteo-api/data/ -type f -wholename "/var/lib/openmeteo-api/data/omfile-*" -name "*_*" -not -wholename "/var/lib/openmeteo-api/data/omfile-dem90/*" -mtime +100 -delete
```

In the future, clean up might be integrated into individual downloaders, or as a separate command.

**Reply by carminelau** [reply]
Hi Patrick, can you detail the various pieces of the controls what they are for ? Especially second.

Because I want to delete data older than two days but I am not understanding how I can do it.
Thanks

**Reply by patrick-zippenfenig** [reply]
It wont be possible to delete data older than 2 days. Each file contains typically data for ~2 weeks. Therefore it can only be deleted in chunks of 2 weeks. Note: It is not strictly 14 days. Depending on the weather model, different chunk sizes are used.

Within the `find` command, you can use `-mtime +2` to delete any database files that have not been updated for 2 days. Therefore you will have data from the past 2-16 days.
